File: Palace/OPDS/TPPOPDSEntryGroupAttributes.m
#import "TPPOPDSEntryGroupAttributes.h"

@interface TPPOPDSEntryGroupAttributes ()

@property (nonatomic) NSURL *href;
@property (nonatomic) NSString *title;

@end

@implementation TPPOPDSEntryGroupAttributes

- (instancetype)initWithHref:(NSURL *const)href title:(NSString *const)title
{
  self = [super init];
  if(!self) return nil;
  
  if(!title) {
    @throw NSInvalidArgumentException;
  }
  
  self.href = href;
  self.title = [title copy];
  
  return self;
}

@end

File: Palace/OPDS/TPPOPDSEntryGroupAttributes.swift
import Foundation

@objc class TPPOPDSEntryGroupAttributes: NSObject {

  @objc private(set) var href: URL?
  @objc private(set) var title: String

  @objc init(href: URL?, title: String) {
    self.title = title
    self.href = href
    super.init()
  }
}
